In [61]:
import pandas as pd

In [62]:
url = 'https://raw.githubusercontent.com/data602sps/datasetspractice/main/nba.csv'
df1 = pd.read_csv(url, index_col=0)
df1

Team  Number Position   Age Height  Weight  \
Name                                                                  
Avery Bradley  Boston Celtics     0.0       PG  25.0    6-2   180.0   
Jae Crowder    Boston Celtics    99.0       SF  25.0    6-6   235.0   
John Holland   Boston Celtics    30.0       SG  27.0    6-5   205.0   
R.J. Hunter    Boston Celtics    28.0       SG  22.0    6-5   185.0   
Jonas Jerebko  Boston Celtics     8.0       PF  29.0   6-10   231.0   
...                       ...     ...      ...   ...    ...     ...   
Shelvin Mack        Utah Jazz     8.0       PG  26.0    6-3   203.0   
Raul Neto           Utah Jazz    25.0       PG  24.0    6-1   179.0   
Tibor Pleiss        Utah Jazz    21.0        C  26.0    7-3   256.0   
Jeff Withey         Utah Jazz    24.0        C  26.0    7-0   231.0   
NaN                       NaN     NaN      NaN   NaN    NaN     NaN   

                         College     Salary  
Name                                         
Avery Bradley              Texas  7730337.0  
Jae Crowder            Marquette  6796117.0  
John Holland   Boston University        NaN  
R.J. Hunter        Georgia State  1148640.0  
Jonas Jerebko                NaN  5000000.0  
...                          ...        ...  
Shelvin Mack              Butler  2433333.0  
Raul Neto                    NaN   900000.0  
Tibor Pleiss                 NaN  2900000.0  
Jeff Withey               Kansas   947276.0  
NaN                          NaN        NaN  

[458 rows x 8 columns]

There are 458 rows and 8 columns for df1.

In [63]:
df1.shape

(458, 8)

In [64]:
df1.dtypes

Team         object
Number      float64
Position     object
Age         float64
Height       object
Weight      float64
College      object
Salary      float64
dtype: object

In [65]:
df1.describe()

Number         Age      Weight        Salary
count  457.000000  457.000000  457.000000  4.460000e+02
mean    17.678337   26.938731  221.522976  4.842684e+06
std     15.966090    4.404016   26.368343  5.229238e+06
min      0.000000   19.000000  161.000000  3.088800e+04
25%      5.000000   24.000000  200.000000  1.044792e+06
50%     13.000000   26.000000  220.000000  2.839073e+06
75%     25.000000   30.000000  240.000000  6.500000e+06
max     99.000000   40.000000  307.000000  2.500000e+07

The describe method returns description of the data in the DataFrame.

If the DataFrame contains numerical data, the description contains these information for each column:

count - The number of not-empty values.
mean - The average (mean) value.
std - The standard deviation.
min - the minimum value.
25% - The 25% percentile*.
50% - The 50% percentile*.
75% - The 75% percentile*.
max - the maximum value.

This data has a lot of NaNs which would need to be cleaned up before using. Other than that, it looks pretty good. 

In [66]:
df1.head() 

Team  Number Position   Age Height  Weight  \
Name                                                                  
Avery Bradley  Boston Celtics     0.0       PG  25.0    6-2   180.0   
Jae Crowder    Boston Celtics    99.0       SF  25.0    6-6   235.0   
John Holland   Boston Celtics    30.0       SG  27.0    6-5   205.0   
R.J. Hunter    Boston Celtics    28.0       SG  22.0    6-5   185.0   
Jonas Jerebko  Boston Celtics     8.0       PF  29.0   6-10   231.0   

                         College     Salary  
Name                                         
Avery Bradley              Texas  7730337.0  
Jae Crowder            Marquette  6796117.0  
John Holland   Boston University        NaN  
R.J. Hunter        Georgia State  1148640.0  
Jonas Jerebko                NaN  5000000.0

.head() shows the first five rows of data in the dataframe.

In [67]:
df1.tail()

Team  Number Position   Age Height  Weight College  \
Name                                                                    
Shelvin Mack  Utah Jazz     8.0       PG  26.0    6-3   203.0  Butler   
Raul Neto     Utah Jazz    25.0       PG  24.0    6-1   179.0     NaN   
Tibor Pleiss  Utah Jazz    21.0        C  26.0    7-3   256.0     NaN   
Jeff Withey   Utah Jazz    24.0        C  26.0    7-0   231.0  Kansas   
NaN                 NaN     NaN      NaN   NaN    NaN     NaN     NaN   

                 Salary  
Name                     
Shelvin Mack  2433333.0  
Raul Neto      900000.0  
Tibor Pleiss  2900000.0  
Jeff Withey    947276.0  
NaN                 NaN

.tail() shows the last five rows of data in the dataframe.

In [68]:
url2 = 'https://raw.githubusercontent.com/data602sps/datasetspractice/main/weather.csv'
df2 = pd.read_csv(url2, index_col=0)
df2

origin  year  month   day  hour   temp   dewp  humid  wind_dir  \
1       EWR  2013    1.0   1.0   0.0  37.04  21.92  53.97     230.0   
2       EWR  2013    1.0   1.0   1.0  37.04  21.92  53.97     230.0   
3       EWR  2013    1.0   1.0   2.0  37.94  21.92  52.09     230.0   
4       EWR  2013    1.0   1.0   3.0  37.94  23.00  54.51     230.0   
5       EWR  2013    1.0   1.0   4.0  37.94  24.08  57.04     240.0   
...     ...   ...    ...   ...   ...    ...    ...    ...       ...   
8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14     200.0   
8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51      60.0   
8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51      40.0   
8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34      40.0   
8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68     210.0   

      wind_speed  wind_gust  precip  pressure  visib  
1       10.35702  11.918651     0.0    1013.9   10.0  
2       13.80936  15.891535     0.0    1013.0   10.0  
3       12.65858  14.567241     0.0    1012.6   10.0  
4       13.80936  15.891535     0.0    1012.7   10.0  
5       14.96014  17.215830     0.0    1012.8   10.0  
...          ...        ...     ...       ...    ...  
8715     4.60312   5.297178     0.0       NaN    4.0  
8716     4.60312   5.297178     0.0    1008.1   10.0  
8717     4.60312   5.297178     0.0    1008.5   10.0  
8718     4.60312   5.297178     0.0    1023.9   10.0  
8719     8.05546   9.270062     0.0    1011.9   10.0  

[8719 rows x 14 columns]

There are 8719 rows and 14 columns for df2

In [69]:
df3 = pd.read_csv('world_population.csv')
df3

Rank CCA3            Country           Capital Continent  \
0      36  AFG        Afghanistan             Kabul      Asia   
1     138  ALB            Albania            Tirana    Europe   
2      34  DZA            Algeria           Algiers    Africa   
3     213  ASM     American Samoa         Pago Pago   Oceania   
4     203  AND            Andorra  Andorra la Vella    Europe   
..    ...  ...                ...               ...       ...   
229   226  WLF  Wallis and Futuna          Mata-Utu   Oceania   
230   172  ESH     Western Sahara          El Aaiún    Africa   
231    46  YEM              Yemen             Sanaa      Asia   
232    63  ZMB             Zambia            Lusaka    Africa   
233    74  ZWE           Zimbabwe            Harare    Africa   

     2022 Population  2020 Population  2015 Population  2010 Population  \
0           41128771         38972230         33753499         28189672   
1            2842321          2866849          2882481          2913399   
2           44903225         43451666         39543154         35856344   
3              44273            46189            51368            54849   
4              79824            77700            71746            71519   
..               ...              ...              ...              ...   
229            11572            11655            12182            13142   
230           575986           556048           491824           413296   
231         33696614         32284046         28516545         24743946   
232         20017675         18927715         16248230         13792086   
233         16320537         15669666         14154937         12839771   

     2000 Population  1990 Population  1980 Population  1970 Population  \
0           19542982         10694796         12486631         10752971   
1            3182021          3295066          2941651          2324731   
2           30774621         25518074         18739378         13795915   
3              58230            47818            32886            27075   
4              66097            53569            35611            19860   
..               ...              ...              ...              ...   
229            14723            13454            11315             9377   
230           270375           178529           116775            76371   
231         18628700         13375121          9204938          6843607   
232          9891136          7686401          5720438          4281671   
233         11834676         10113893          7049926          5202918   

     Area (km²)  Density (per km²)  Growth Rate  World Population Percentage  
0        652230            63.0587       1.0257                         0.52  
1         28748            98.8702       0.9957                         0.04  
2       2381741            18.8531       1.0164                         0.56  
3           199           222.4774       0.9831                         0.00  
4           468           170.5641       1.0100                         0.00  
..          ...                ...          ...                          ...  
229         142            81.4930       0.9953                         0.00  
230      266000             2.1654       1.0184                         0.01  
231      527968            63.8232       1.0217                         0.42  
232      752612            26.5976       1.0280                         0.25  
233      390757            41.7665       1.0204                         0.20  

[234 rows x 17 columns]

There are 234 rows and 17 columns for df3

In [70]:
df4 = pd.read_csv('netflix_titles.csv')
df4

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

There are 8807 rows and 12 columns for df4

Extra Credit:

In [71]:
extracredit = pd.read_csv('world_population.csv', header = None)

In [72]:
extracredit.head()

0     1               2          3          4                5   \
0  Rank  CCA3         Country    Capital  Continent  2022 Population   
1    36   AFG     Afghanistan      Kabul       Asia         41128771   
2   138   ALB         Albania     Tirana     Europe          2842321   
3    34   DZA         Algeria    Algiers     Africa         44903225   
4   213   ASM  American Samoa  Pago Pago    Oceania            44273   

                6                7                8                9   \
0  2020 Population  2015 Population  2010 Population  2000 Population   
1         38972230         33753499         28189672         19542982   
2          2866849          2882481          2913399          3182021   
3         43451666         39543154         35856344         30774621   
4            46189            51368            54849            58230   

                10               11               12          13  \
0  1990 Population  1980 Population  1970 Population  Area (km²)   
1         10694796         12486631         10752971      652230   
2          3295066          2941651          2324731       28748   
3         25518074         18739378         13795915     2381741   
4            47818            32886            27075         199   

                  14           15                           16  
0  Density (per km²)  Growth Rate  World Population Percentage  
1            63.0587       1.0257                         0.52  
2            98.8702       0.9957                         0.04  
3            18.8531       1.0164                         0.56  
4           222.4774       0.9831                            0

In [73]:
extracreditwheader = pd.read_csv('world_population.csv')

In [74]:
extracreditwheader.head()

Rank CCA3         Country           Capital Continent  2022 Population  \
0    36  AFG     Afghanistan             Kabul      Asia         41128771   
1   138  ALB         Albania            Tirana    Europe          2842321   
2    34  DZA         Algeria           Algiers    Africa         44903225   
3   213  ASM  American Samoa         Pago Pago   Oceania            44273   
4   203  AND         Andorra  Andorra la Vella    Europe            79824   

   2020 Population  2015 Population  2010 Population  2000 Population  \
0         38972230         33753499         28189672         19542982   
1          2866849          2882481          2913399          3182021   
2         43451666         39543154         35856344         30774621   
3            46189            51368            54849            58230   
4            77700            71746            71519            66097   

   1990 Population  1980 Population  1970 Population  Area (km²)  \
0         10694796         12486631         10752971      652230   
1          3295066          2941651          2324731       28748   
2         25518074         18739378         13795915     2381741   
3            47818            32886            27075         199   
4            53569            35611            19860         468   

   Density (per km²)  Growth Rate  World Population Percentage  
0            63.0587       1.0257                         0.52  
1            98.8702       0.9957                         0.04  
2            18.8531       1.0164                         0.56  
3           222.4774       0.9831                         0.00  
4           170.5641       1.0100                         0.00

In [135]:
messy = pd.read_csv('messy-data.csv')

In [136]:
messy

Unnamed: 0 Seasonal rainfall in Lake Victoria and Simiyu     Unnamed: 2  \
0          NaN                                           NaN            NaN   
1          NaN                                 Month, period  Lake Victoria   
2          NaN                                 Jan,2001-2019        3.176mm   
3          NaN                                 Feb,2001-2019        3.477mm   
4          NaN                                 Mar,2001-2019          4.687   
5          NaN                                 Apr,2001-2019          7.005   
6          NaN                                 May,2001-2019          9.363   
7          NaN                                 Jun,2001-2019          3.430   
8          NaN                                 Jul,2001-2019          1.764   
9          NaN                                 Aug,2001-2019          2.813   
10         NaN                                 Sep,2001-2019          3.979   
11         NaN                                 Oct,2001-2019          5.318   
12         NaN                                 Nov,2001-2019          5.118   
13         NaN                                 Dec,2001-2019          4.168   
14         NaN                                           NaN            NaN   
15         NaN                                           NaN            NaN   
16         NaN                                           NaN            NaN   
17         NaN                                           NaN            NaN   
18         NaN                                           NaN            NaN   
19         NaN                                           NaN            NaN   

   Unnamed: 3  Unnamed: 4  Unnamed: 5  Unnamed: 6  Unnamed: 7  Unnamed: 8  \
0         NaN         NaN         NaN         NaN         NaN         NaN   
1      Simiyu         NaN         NaN         NaN         NaN         NaN   
2       2.908         NaN         NaN         NaN         NaN         NaN   
3       1.8mm         NaN         NaN         NaN         NaN         NaN   
4       2.981         NaN         NaN         NaN         NaN         NaN   
5       4.754         NaN         NaN         NaN         NaN         NaN   
6       4.077         NaN         NaN         NaN         NaN         NaN   
7       1.047         NaN         NaN         NaN         NaN         NaN   
8       0.195         NaN         NaN         NaN         NaN         NaN   
9       0.334         NaN         NaN         NaN         NaN         NaN   
10      1.206         NaN         NaN         NaN         NaN         NaN   
11      2.455         NaN         NaN         NaN         NaN         NaN   
12      3.091         NaN         NaN         NaN         NaN         NaN   
13      3.890         NaN         NaN         NaN         NaN         NaN   
14        NaN         NaN         NaN         NaN         NaN         NaN   
15        NaN         NaN         NaN         NaN         NaN         NaN   
16        NaN         NaN         NaN         NaN         NaN         NaN   
17        NaN         NaN         NaN         NaN         NaN         NaN   
18        NaN         NaN         NaN         NaN         NaN         NaN   
19        NaN         NaN         NaN         NaN         NaN         NaN   

    Unnamed: 9  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN  
5          NaN  
6          NaN  
7          NaN  
8          NaN  
9          NaN  
10         NaN  
11         NaN  
12         NaN  
13         NaN  
14         NaN  
15         NaN  
16         NaN  
17         NaN  
18         NaN  
19         NaN

In [137]:
messy.columns

Index(['Unnamed: 0', 'Seasonal rainfall in Lake Victoria and Simiyu',
       'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6',
       'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9'],
      dtype='object')

In [138]:
messy.drop(messy.columns[[0, 4, 5, 6, 7, 8, 9]], axis=1, inplace=True)

In [139]:
messy

Seasonal rainfall in Lake Victoria and Simiyu     Unnamed: 2 Unnamed: 3
0                                            NaN            NaN        NaN
1                                  Month, period  Lake Victoria     Simiyu
2                                  Jan,2001-2019        3.176mm      2.908
3                                  Feb,2001-2019        3.477mm      1.8mm
4                                  Mar,2001-2019          4.687      2.981
5                                  Apr,2001-2019          7.005      4.754
6                                  May,2001-2019          9.363      4.077
7                                  Jun,2001-2019          3.430      1.047
8                                  Jul,2001-2019          1.764      0.195
9                                  Aug,2001-2019          2.813      0.334
10                                 Sep,2001-2019          3.979      1.206
11                                 Oct,2001-2019          5.318      2.455
12                                 Nov,2001-2019          5.118      3.091
13                                 Dec,2001-2019          4.168      3.890
14                                           NaN            NaN        NaN
15                                           NaN            NaN        NaN
16                                           NaN            NaN        NaN
17                                           NaN            NaN        NaN
18                                           NaN            NaN        NaN
19                                           NaN            NaN        NaN

In [140]:
messy=messy.drop(messy.index[14:20])

In [141]:
messy

Seasonal rainfall in Lake Victoria and Simiyu     Unnamed: 2 Unnamed: 3
0                                            NaN            NaN        NaN
1                                  Month, period  Lake Victoria     Simiyu
2                                  Jan,2001-2019        3.176mm      2.908
3                                  Feb,2001-2019        3.477mm      1.8mm
4                                  Mar,2001-2019          4.687      2.981
5                                  Apr,2001-2019          7.005      4.754
6                                  May,2001-2019          9.363      4.077
7                                  Jun,2001-2019          3.430      1.047
8                                  Jul,2001-2019          1.764      0.195
9                                  Aug,2001-2019          2.813      0.334
10                                 Sep,2001-2019          3.979      1.206
11                                 Oct,2001-2019          5.318      2.455
12                                 Nov,2001-2019          5.118      3.091
13                                 Dec,2001-2019          4.168      3.890

In [142]:
messy=messy.drop(messy.index[0:2])

In [143]:
messy

Seasonal rainfall in Lake Victoria and Simiyu Unnamed: 2 Unnamed: 3
2                                  Jan,2001-2019    3.176mm      2.908
3                                  Feb,2001-2019    3.477mm      1.8mm
4                                  Mar,2001-2019      4.687      2.981
5                                  Apr,2001-2019      7.005      4.754
6                                  May,2001-2019      9.363      4.077
7                                  Jun,2001-2019      3.430      1.047
8                                  Jul,2001-2019      1.764      0.195
9                                  Aug,2001-2019      2.813      0.334
10                                 Sep,2001-2019      3.979      1.206
11                                 Oct,2001-2019      5.318      2.455
12                                 Nov,2001-2019      5.118      3.091
13                                 Dec,2001-2019      4.168      3.890

In [144]:
messy.set_index('Seasonal rainfall in Lake Victoria and Simiyu', inplace = True)

In [145]:
messy.head()

Unnamed: 2 Unnamed: 3
Seasonal rainfall in Lake Victoria and Simiyu                      
Jan,2001-2019                                    3.176mm      2.908
Feb,2001-2019                                    3.477mm      1.8mm
Mar,2001-2019                                      4.687      2.981
Apr,2001-2019                                      7.005      4.754
May,2001-2019                                      9.363      4.077

In [146]:
messy.columns = ['Lake Victoria Seasonal Rainfall in mm', 'Simiyu Seasonal Rainfall in mm']

In [147]:
messy

Lake Victoria Seasonal Rainfall in mm  \
Seasonal rainfall in Lake Victoria and Simiyu                                         
Jan,2001-2019                                                               3.176mm   
Feb,2001-2019                                                               3.477mm   
Mar,2001-2019                                                                 4.687   
Apr,2001-2019                                                                 7.005   
May,2001-2019                                                                 9.363   
Jun,2001-2019                                                                 3.430   
Jul,2001-2019                                                                 1.764   
Aug,2001-2019                                                                 2.813   
Sep,2001-2019                                                                 3.979   
Oct,2001-2019                                                                 5.318   
Nov,2001-2019                                                                 5.118   
Dec,2001-2019                                                                 4.168   

                                              Simiyu Seasonal Rainfall in mm  
Seasonal rainfall in Lake Victoria and Simiyu                                 
Jan,2001-2019                                                          2.908  
Feb,2001-2019                                                          1.8mm  
Mar,2001-2019                                                          2.981  
Apr,2001-2019                                                          4.754  
May,2001-2019                                                          4.077  
Jun,2001-2019                                                          1.047  
Jul,2001-2019                                                          0.195  
Aug,2001-2019                                                          0.334  
Sep,2001-2019                                                          1.206  
Oct,2001-2019                                                          2.455  
Nov,2001-2019                                                          3.091  
Dec,2001-2019                                                          3.890

In [148]:
messy.index.names = ['Date']

In [150]:
messy['Lake Victoria Seasonal Rainfall in mm'] = messy['Lake Victoria Seasonal Rainfall in mm'].map(lambda x: x.rstrip('m'))

In [152]:
messy['Simiyu Seasonal Rainfall in mm'] = messy['Simiyu Seasonal Rainfall in mm'].map(lambda x: x.rstrip('m'))

In [155]:
messy['Lake Victoria Seasonal Rainfall in mm'] = messy['Lake Victoria Seasonal Rainfall in mm'].astype('float')

In [156]:
messy['Simiyu Seasonal Rainfall in mm'] = messy['Simiyu Seasonal Rainfall in mm'].astype('float')

In [157]:
messy

Lake Victoria Seasonal Rainfall in mm  \
Date                                                   
Jan,2001-2019                                  3.176   
Feb,2001-2019                                  3.477   
Mar,2001-2019                                  4.687   
Apr,2001-2019                                  7.005   
May,2001-2019                                  9.363   
Jun,2001-2019                                  3.430   
Jul,2001-2019                                  1.764   
Aug,2001-2019                                  2.813   
Sep,2001-2019                                  3.979   
Oct,2001-2019                                  5.318   
Nov,2001-2019                                  5.118   
Dec,2001-2019                                  4.168   

               Simiyu Seasonal Rainfall in mm  
Date                                           
Jan,2001-2019                           2.908  
Feb,2001-2019                           1.800  
Mar,2001-2019                           2.981  
Apr,2001-2019                           4.754  
May,2001-2019                           4.077  
Jun,2001-2019                           1.047  
Jul,2001-2019                           0.195  
Aug,2001-2019                           0.334  
Sep,2001-2019                           1.206  
Oct,2001-2019                           2.455  
Nov,2001-2019                           3.091  
Dec,2001-2019                           3.890